setuptools/tests/test_config.py

```diff
@@ -4,18 +4,23 @@
 from mock import patch
 from setuptools.dist import Distribution, _Distribution
 from setuptools.config import ConfigHandler, read_configuration
+from setuptools.extern.six import PY2, PY3
 
+py2_only = pytest.mark.xfail(not PY2, reason="Test runs on Python 2 only")
```

Comment:
I think these should be `pytest.mark.skip`, not `xfail`.

Reply:
will have to fix the tests for find_packages_ns as well. you would want to fix the other tests that use that pattern.

Reply:
I concur.

Reply:
six.PY2 seems to fail when marking tests for specific versions

```
SKIP [1] setuptools/tests/test_config.py:604: Test runs on Python 2 only
```

Reply:
I think that's indicating that the test was skipped. When you mark something `xfail`, it actually *runs* the test and ignores the result if it fails (or if you make it a strict xfail, actually fails the test suite if the test passes). `skip` doesn't run the test at all, so I believe that message is expected.

You are right that we should be switching many `xfail`s over to `skip`. I was planning on doing that at some point, but it takes time to figure out which ones are intended as an actual `xfail` (which generally means "this should work but it's a known bug - when this test starts passing drop this decorator") and which are just using the wrong decorator for `skip`.